src/gps/dateTime.c

```diff
@@ -133,3 +133,118 @@ tiny_millis_t getUptime() {
 int getUptimeAsInt() {
    return (int) getUptime();
 }
+
+void getDateTimeFromEpochMillis(DateTime *dateTime, millis_t millis) {
```

Comment:
TIL that [Return Value Optimization](http://en.wikipedia.org/wiki/Return_value_optimization) apparently only applies to C++ and not C.  So you are correct in noting that there is a 2x usage in stack space.  I was of the opinion that our optimizer would apply this and thus make the space argument moot.

In general I favor simple expressions over complex ones.  Primarily I do this not for myself, but for the other programmers that need to read and use the code that I am creating.  Having a method say its going to do something and return the result is much easier to grasp than needing to understand that you need to pass your return value by reference into the method so it may be populated.  I also follow this strategy to isolate memory access between functions.  Passing pointers is both wonderful and the work of satan when it breaks.  Following a memory isolationist strategy tends to lead to fewer memory issues because you isolate memory usage at the functional level.  This often buys you better reliability at the expense of more memory usage.  This in my book is a worthwhile tradeoff in our case.

  Sure the struct we are passing in this particular example is large when compared to other structures in our code base, but in the overall scheme of things it is still quite small.  At what point do we start getting this anal retentive with stack space savings?  Do we not pass a unit64 type back on the stack because it would cause us to use 16 bytes instead of 8?  What about a structure that is 16 bytes in size?  Where do you draw the tradeoff between code that is more simple and less prone to screw ups vs speed gains and stack space savings?

I don't think your arguments are wrong.  In fact they are spot on accurate in terms of space usage.  We absolutely need to be careful about how much stack space we use.  But I feel that what you are doing is a premature optimization when one is not really needed.  We have more than enough space to spare in our stack sizes and we are very good at keeping our structures small and to the point.  I would trade a bit of memory waste for cleaner more isolated code any day.  All within reason of course.